services/github/github-total-star.service.js

```diff
@@ -0,0 +1,259 @@
+</p>
+`
+const orgDocumentation = `${commonDocumentation}
+<p>
+  <b>Note:</b> ${customDocumentation}
+</p>`
+
+const pageInfoSchema = Joi.object({
+  hasNextPage: Joi.boolean().required(),
+  endCursor: Joi.string().allow(null).required(),
+}).required()
+
+const nodesSchema = Joi.array()
+  .items(
+    Joi.object({
+      stargazers: Joi.object({
+        totalCount: nonNegativeInteger,
+      }).required(),
+    })
+  )
+  .default([])
+
+const repositoriesSchema = Joi.object({
+  pageInfo: pageInfoSchema,
+  nodes: nodesSchema,
+}).required()
+
+const schema = Joi.object({
+  data: Joi.alternatives(
+    Joi.object({
+      user: Joi.object({
+        repositories: repositoriesSchema,
+      }).required(),
+    }).required(),
+    Joi.object({
+      organization: Joi.object({
+        repositories: repositoriesSchema,
+      }).required(),
+    }).required()
+  ).required(),
+}).required()
+
+const userQuery = gql`
+  query fetchStars(
+    $user: String!
+    $nextCursor: String
+    $affiliations: [RepositoryAffiliation]!
+  ) {
+    user(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        ownerAffiliations: $affiliations
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+
+const orgQuery = gql`
+  query fetchStars($user: String!, $nextCursor: String) {
+    organization(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+const affiliationsAllowedValues = [
+  'OWNER',
+  `COLLABORATOR`,
+  'ORGANIZATION_MEMBER',
+]
+/**
+ * Validates affiliations should contain combination of allowed values in any order.
+ *
+ * @param {string} value affiliation current value
+ * @param {*} helpers object to construct custom error
+ *
+ * @returns {string} valiadtion error or value unchanged
+ */
+const validateAffiliations = (value, helpers) => {
+  const values = value.split(',')
+  if (values.some(e => !affiliationsAllowedValues.includes(e))) {
+    return helpers.error('any.invalid')
+  }
+  return value
+}
+
+const queryParamSchema = Joi.object({
+  org: Joi.valid('').optional(),
+  affiliations: Joi.string().default('OWNER').custom(validateAffiliations),
+}).required()
+
+module.exports = class GithubTotalStarService extends GithubAuthV4Service {
+  static get defaultLabel() {
+    return 'Stars'
+  }
+
+  static get category() {
+    return 'social'
+  }
+
+  static get route() {
+    return {
+      base: 'github/stars',
+      pattern: ':user',
+      queryParamSchema,
+    }
+  }
```

Comment:
Let's use the new static props that we're converting to so that we can avoid adding rework

```suggestion
  static defaultLabel = 'Stars'
  static category = 'social'
  static route = {
      base: 'github/stars',
      pattern: ':user',
      queryParamSchema,
  }
```